### ISSUE TITLE
[Bug]: Respect config during initialization

### ISSUE THREAD

**Issue by jemag** [mrjones2014/smart-splits.nvim#270]
### Similar Issues

- [X] Before filing, I have searched for similar issues.

### Neovim Version

NVIM v0.11.0-dev-1369+gb03e790cdd

### Multiplexer Integration

tmux

### Multiplexer Version

tmux 3.5a

### Steps to Reproduce

1. Create config for `smart-splits.nvim`
2. Open neovim and notice config is ignored at initialization

### Expected Behavior

Config is respected

### Actual Behavior

Config is ignored

### Minimal Configuration to Reproduce

See additional details.

### Additional Details and/or Screenshots

Currently increasing the `log_level` or disabling the `multiplexer_integration` is ignored during the initialization phase, as the user config is only loaded afterwards.

For example, this config
```lua
require('smart-splits').setup({
  -- enable or disable the tmux Integration
  multiplexer_integration = false,
  log_level = 'error',
  cursor_follows_swapped_bufs = true,
})
```
Will not be respected on initial loading of neovim.

This causes issues where despite increasing the `log_level` and disabling the `multiplexer_integration`, I will still receive error messages based on the tmux integration (which I do not wish to use).
In my case, I use the following for quick editing of tmux scroll back buffer in neovim
```text
bind-key e 'capture-pane -JS -1000 ; save-buffer ~/tmux.history ; delete-buffer; display-popup -h 75% -w 75% -E $EDITOR ~/tmux.history'
```
However this will always trigger an annoying logging message since initially the config is ignored and tmux is initialized. Basically tmux init happens before config is loaded. Here is the sequence of event when loading the scrollback buffer:
```text
DEBUGPRINT[7]: init.lua:41: require('smart-splits.config').log_level="info"
DEBUGPRINT[7]: init.lua:41: require('smart-splits.config').log_level="info"
[smart-splits.nvim] 
tmux init: could not detect pane ID!
DEBUGPRINT[8]: config.lua:114: new_config={
  cursor_follows_swapped_bufs = true,
  log_level = "error",
  multiplexer_integration = false
}
```

It would be great if the config can be respected and prevent tmux initialization and therefore error message.

**Comment by mrjones2014** [comment]
Does #271 solve your issue? Basically uses the `VimEnter` autocmd so that (hopefully) your `setup()` call happens before the rest of the init logic.

**Comment by jemag** [comment]
unfortunately it does not:
![image](https://github.com/user-attachments/assets/3040f504-1248-4c76-a45e-db0f494eba20)
would it not be possible to simply initialize it after setup is called?

Is this so that users are not required to call the `setup` function?

**Comment by mrjones2014** [comment]
Yeah, the idea is so that you don't necessarily need to call `setup() `. I'm potentially willing to revisit that, but I'll need to put more thought into it since it would be a breaking change.

Likely won't get to it till after christmas since I'm at my parents' house till then.

**Comment by mrjones2014** [comment]
I think I've found a solution that doesn't require a breaking change. Could you test #279? tl;dr: you can set `vim.g.smart_splits_multiplexer_integration` any time before the plugin is loaded.

**Comment by jemag** [comment]
sorry been really busy, will test it tonight or tomorrow

**Comment by jemag** [comment]
Seems to be working well, thanks!

Although perhaps worth mentioning in the docs that the value should be `false`, since a lot of these global values in vim often use the `0` or `1` instead:

![Image](https://github.com/user-attachments/assets/2f6a2221-4340-405f-ab80-dc7358e78d81)
 which won't work in this case.

**Comment by mrjones2014** [comment]
I can make it work with either one.
